src/wwwroot/template-editor/adapterService.js

```diff
@@ -0,0 +1,356 @@
+(function() {
+    'use strict';
+  
+    angular
+      .module('mseditor')
+      .service('adapterService', adapterService);
+  
+      adapterService.$inject = [
+      '$http',
+      '$q'
+    ];
+  
+    function adapterService($http, $q) {
+      var service = {
+        getCampaign: getCampaign,
+        campaignSaveChanges: campaignSaveChanges,
+        campaignSaveChangesAsTemplate: campaignSaveChangesAsTemplate,
+        getCampaignRssPreview: getCampaignRssPreview,
+        deleteImages: deleteImages,
+        getImagesForCampaign: getImagesForCampaign,
+        getPreviewImage: getPreviewImage,
+        getPreviewImageWithPlay: getPreviewImageWithPlay,
+        getTiendaNubeProducts: getTiendaNubeProducts,
+        getSettings: getSettings
+      };
+  
+      return service;
+
+      /**
+       * Return a preview placeholder image
+       * @method
+       * @argument {object} params
+       * @example
+       * {
+       *  fileName: fileName,
+       *  providerName: providerName,
+       *  idCampaign: campaign.getIdCampaign(),
+       *  idTemplate: campaign.getIdTemplate()
+       * }
+       */
+      function getPreviewImageWithPlay(params) {
+        console.log("getPreviewImageWithPlay");
+        var response = {
+          ThumbnailUrlWithPlay: 'https://i.vimeocdn.com/filter/overlay?src0=' + 'http://www.wpclipart.com/travel/US_Road_Signs/info/temporary.png' + '&src1=http%3A%2F%2Ff.vimeocdn.com%2Fp%2Fimages%2Fcrawler_play.png'
+        };
+        var fakePromise = $q.resolve(response);
+        return fakePromise;
+      }
+
+      /**
+       * Return a preview image from a video
+       * @method
```

Comment:
Is _@method_ tag necessary? It seems that it is not being used too much in _JSDoc_ examples.

Reply:
@method is used to describe functions that has being part of a object property, and all of this methods are being part of the service object of the serviceAdapter

Reply:
I understand, but, for example [here](http://usejsdoc.org/howto-es2015-classes.html) the scenario is very similar and they omit that. 

If it does not give us any benefit, please remove.

Reply:
No hurry, but do not forget to answer me this question.

Reply:
I guess it is not a mandatory guideline, we can remove it.